README.md

```diff
@@ -1,122 +1,570 @@
+So in the case you plan to make changes to the operator helm_release once deployed, run preliminary a `terraform plan` to be sure that the release isn't destroyed and recreated.
+
+### Examples of the secrets format and layout
+
+<details><summary>dockerconfigjson from arbitrary or iam_credentials</summary>
+
+Secret Body >
+
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dockerconfigjson-iam
+  namespace: test-ns
+data:
+  .dockerconfigjson: [BASE64ENCODED]
+type: kubernetes.io/dockerconfigjson
+```
+
+Base64 Decoded `.dockerconfigson` >
+
+```
+{
+  "auths": {
+    "us.icr.io": {
+      "username": "iamapikey",
+      "password": "APIKEYVALUE", # pragma: allowlist secret
+      "email": "terraform@ibm.com"
+    }
+  }
+}
+```
+
+</details>
+
+<details><summary>dockerconfigjson from username_password</summary>
+
+Secret Body >
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: dockerconfigjson-uc
+  namespace: test-ns
+data:
+  .dockerconfigjson: [BASE64ENCODED]
+type: kubernetes.io/dockerconfigjson
+```
+
+Base64 Decoded `.dockerconfigson` >
+
+```
+{
+  "auths": {
+    "xx.artifactory.xyz-devops.com": {
+      "username": "artifactoryuser@org.com",
+      "password": "APIKEYVALUE" # pragma: allowlist secret
+    }
+  }
+}
+```
+
+</details>
+
+<details><summary>opaque from arbitrary or iam_credentials</summary>
+
+Secret Body >
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: opaque-arb
+  namespace: test-ns
+type: Opaque
+data:
+  apikey: APIKEYVALUE # pragma: allowlist secret
+
+```
+
+</details>
+
+<details><summary>opaque from username_password</summary>
+
+Secret Body >
+```
+apiVersion: v1
+kind: Secret
+metadata:
+  name: opaque-uc
+  namespace: test-ns
+type: Opaque
+data:
+  username: test-user
+  password: PASSWORDVALUE # pragma: allowlist secret
+
+```
+
+</details>
+
+
+NIST controls do not apply to this module.
+
+## Usage
+
+```hcl
+# Replace "master" with a GIT release version to lock into a specific release
+module "es_kubernetes_secret" {
+  source                     = "https://github.com/terraform-ibm-modules/terraform-ibm-external-secrets-operator.git//modules/eso-external-secret?ref=master"
+  es_kubernetes_secret_type = "dockerconfigjson"
+  sm_secret_type = "iam_credentials"
+  sm_secret_id = module.docker_config.serviceid_apikey_secret_id
+  eso_setup = true
+  es_kubernetes_namespaces = var.es_kubernetes_namespaces
+  es_docker_email = "terraform@ibm.com"
+  eso_generic_secret_apikey = data.ibm_secrets_manager_secret.secret_puller_secret.api_key # pragma: allowlist secret
+  secrets_manager_guid = module.secrets_manager_iam_configuration.secrets_manager_guid
+  region = "us-south"
+  es_kubernetes_secret_name = "dockerconfigjson-iam"
+  depends_on = [
+    kubernetes_namespace.cluster_namespaces
+  ]
+  es_kubernetes_secret_data_key = "apiKey"
+  es_helm_rls_name = "es-docker-iam"
+}
+```
+
+<!-- BEGIN EXAMPLES HOOK -->
```

Comment:
Examples hook has been replaced by the overview hook - it should not be used anymore